corehq/apps/app_manager/static/app_manager/js/report-module.js

```diff
@@ -89,7 +89,95 @@ var ReportModule = (function () {
         this.allGraphTypes = ['bar', 'time', 'xy'];
     }
 
-    function ReportConfig(report_id, display, availableReportIds, reportCharts, graph_configs, language) {
+    function FilterConfig(report_id, reportId, filterValues, reportFilters, changeSaveButton) {
+        var self = this;
+
+        this.reportFilters = JSON.parse(JSON.stringify(reportFilters)) || {};
```

Comment:
If reportFilters ever going to be undefined? If so, will stringify fail? If it'll always at least be an empty object, won't stringify make it `"{}"` and then parse make it `{}` so you don't need the `|| {}`?